typing/env.ml

```diff
@@ -638,14 +638,15 @@ let components_of_module_maker' =
             (module_components_repr, module_components_failure) result)
 
 let components_of_functor_appl' =
-  ref ((fun ~loc:_ _f _env _p1 _p2 -> assert false) :
-          loc:Location.t -> functor_components -> t ->
```

Comment:
Because a function `... -> Path.t -> functor_components -> Path.t -> ....` causes lot's of swearing-induced debugging because I constantly swapped the two paths. The tuples makes it very clear which path is associated to the functor component and avoid all issues.
I tested with labels, it's strictly less clear and more verbose.

Finally, I think we can agree than a single short-lived tuple allocation is pretty negligeble compared to "compute the components of this functor application" and is a very low cost for absence of horrible to debug errors.

Reply:
This is now fixed.